### DISCUSSION TITLE
URL Tags

### DISCUSSION THREAD

**Discussion by JosephTLyons** [rwf2/Rocket#1997] [Questions]
In [Django](https://www.djangoproject.com), you can put URL tags in your HTML files like this

When registering URLs in Django, you can give them a name

`path('<url_path>', SomeView.as_view(), name='<url_name>')`

Then, inside of your HTML file, you can use a [URL tag](https://docs.djangoproject.com/en/3.2/ref/templates/builtins/#url) like this:

`{% url <url_name> $}`

Is this possible in rocket.rs using Tera templating engine?  I can hardcode URLs into HTML files, but it makes changing routes a bit messier.  I couldn't find anything like this in the documentation for either rocket.rs or Tera, but maybe I missed something.

**Comment by edgarogh** [comment]
Rocket has a [`uri!`](https://docs.rs/rocket/latest/rocket/macro.uri.html) macro designed specifically for that, but it's a Rust macro (not callable from Tera since Tera templates are dynamic and Rust macros are expanded at compile time). You could however initialize the fairing with [`Template::custom`](https://api.rocket.rs/v0.5-rc/rocket_dyn_templates/struct.Template.html#method.custom) and register a custom Tera function that returns the built URL for your use case.

Maybe something like this could work ?
```rs
rocket::build()
    .attach(Template::custom(|engines| {
        engines.tera.register_function("build_profile_uri", |args| Ok(uri!(...).into()))
    }))
    .mount(...)
```
